modules/drivers/redshift/test/metabase/driver/redshift_test.clj

```diff
@@ -479,3 +481,105 @@
                        qual-tbl-partial-select-name
                        qual-tbl-partial-update-name
                        username)))))))))
+
+(defn- detect-canceled-queries-test-create-sleep-function!
+  "Create a redshift_test__sleep() function since Redshift doesn't have a pg_sleep() function"
+  []
+  (sql-jdbc.execute/do-with-connection-with-options
+   :redshift
+   (driver/with-driver :redshift (mt/db))
+   {:write? true}
+   (fn [^java.sql.Connection conn]
+     (let [sql (str/join \newline ["CREATE OR REPLACE FUNCTION redshift_test__sleep(seconds integer)"
+                                   "RETURNS integer immutable AS $$"
+                                   "  import time"
+                                   "  time.sleep(seconds)"
+                                   "  return 0"
+                                   "$$ LANGUAGE plpythonu;"])]
+       (log/infof "[redshift]\n%s" sql)
+       (next.jdbc/execute! conn [sql])))))
+
+(defn- detect-canceled-queries-test-start-long-running-query!
+  "Start a long-running query in a background thread."
+  []
+  (mt/test-driver :redshift
```

Comment:
Yes because it binds `metabase.driver/*driver*` and also won't run the test if you're running locally and have this file on your classpath but have `DRIVERS=` set to something that doesn't include `:redshift`

Reply:
Sorry, I should have been more explicit -- I thought of it as follows: `mt/test-driver :redshift` is used right at the beginning of `detect-canceled-queries-test` and I believe this function is supposed to run only during that test.

That's why I thought it may be redundant.